spec/rubocop/cop/rspec/nested_groups_spec.rb

```diff
@@ -20,6 +20,36 @@
     RUBY
   end
 
+  it 'flags nested example groups' do
+    expect_offense(<<-RUBY)
+      example_group MyClass do
+        context 'when foo' do
+          context 'when bar' do
+            context 'when baz' do
+            ^^^^^^^^^^^^^^^^^^ Maximum example group nesting exceeded [4/3].
+            end
+          end
+        end
+      end
+    RUBY
+  end
+
+  it 'flags nested example groups inside shared examples' do
+    expect_offense(<<-RUBY)
+      shared_examples_for 'nested like express' do
+        context 'when foo' do
+          context 'when bar' do
+            context 'when baz' do
+              context 'when qux' do
+              ^^^^^^^^^^^^^^^^^^ Maximum example group nesting exceeded [4/3].
```

Comment:
`shared_examples_for` is not counted.
If we count it in, even one of our own specs triggers an offence:
```ruby
RSpec.describe RuboCop::Cop::RSpec::PredicateMatcher, :config do
  context 'when enforced style is `explicit`' do
    shared_examples 'explicit common' do
      context 'when custom matchers are allowed' do
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Maximum example group nesting exceeded [4/3].
```